frontend/packages/console-plugin-sdk/README.md

```diff
@@ -0,0 +1,234 @@
+
+- `ModelDefinition` - Add new k8s model definitions.
+- `FeatureFlag/Model` (in future also FeatureFlag/Action) - Add new feature flag, and use the newly added feature flags (possibly together with core Console flags) to gate its extensions.
+
+``` bash
+    .
+    ├── ...
+    ├── frontend
+    │   ├── packages
+    |   |   ├── ...
+    │   |   ├── my-plugin
+    |   |   |   ├── integration-tests
+    |   |   |   |   └── ...
+    |   |   |   ├── src
+    |   |   |   |   ├── plugin.tsx
+    |   |   |   |   ├── __tests__
+    |   |   |   |   |   ├── some-test.spec.ts
+    |   |   |   |   |   └── ...
+    |   |   |   |   └── ...
+    |   |   |   ├── OWNERS
+    |   |   |   ├── package.json
+    |   |   |   └── ...
+    |   |   └── ...
+    │   └── ...
+    └── ...
+```
+
+## Plugin Directory
+
+The plugin directory name should match the plugin name (core plugins are an exception), as a convention plugin
+directory names should end with the `-plugin` suffix (eg. `my-plugin`).
+
+Packages whose name starts with `console-` are expected to be maintained by core Console team, so they shouldn't have an `OWNERS` file.
+Every (non-core) plugin package must have an `OWNERS` file. Its directory name should end with `-plugin` to denote its purpose.
+
+## Package.json
+
+Plugins are packages. The package definition of a plugin should include a `consolePlugin`
+map that defines the plugin entry module (value of `entry` key), and the plugin integration tests.
+
+The `consolePlugin.entry` path should point to a module that exports the Plugin object (plugin entry module).
+
+From plugin author perspective, a Plugin is simply a list of extensions.
+The plugin entry module is loaded right upon Console startup, so it should lazy-load other code (use the import function) instead of directly referring to it.
+
+``` json
+// File: package.json
+
+{
+  "name": "@console/demo-plugin",
+  "version": "0.0.0-fixed",
+  "description": "Demo plugin for Console web application",
+  "private": true,
+  "dependencies": {
+    "@console/plugin-sdk": "0.0.0-fixed",
+    "@console/shared": "0.0.0-fixed"
+  },
+  "devDependencies": {
+    "@console/internal-integration-tests": "0.0.0-fixed"
+  },
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    "integrationTests": {
+      "demo": [
+        "@console/internal-integration-tests/tests/base.scenario.ts",
+        "integration-tests/tests/demo.scenario.ts"
+      ]
+    }
+  }
+}
+```
+
+## Plugin Entry Module (`.../src/plugin.tsx`)
+
+The Plugin Entry Module is defined in `package.json` in the `consolePlugin.entry` variable,
+relative to the plugin root directory.
+
+``` json
+// File: package.json
+
+{
+  ...
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    ...
+  }
+}
+```
+
+The Plugin Entry Module will export an array of plugin objects of type `Plugin<ConsumedExtensions>`. Each `Plugin` object have a type
+declaring the plugin capabilities and options (for example `Page/Route`, `Dashboards/Card` types).
```

Comment:
"The Plugin Entry Module will export an array of plugin objects of type `Plugin<ConsumedExtensions>`. Each `Plugin` object have a type declaring the plugin capabilities and options (for example `Page/Route`, `Dashboards/Card` types)."

"The plugin entry module will export a variable of type `Plugin<ConsumedExtensions>` which will contain an array of objects of type `Extension`. Each `Extension` object must contain the following properties:

- `type`: A string that describes the broader category and any specialization(s) into which the extension falls 
- `properties`: An object containing options relevant to the extension"